inbm/dispatcher-agent/tests/unit/source/test_ubuntu_source_cmd.py

```diff
@@ -192,14 +193,44 @@ def test_update_sources_os_error(self):
 
 
 class TestUbuntuApplicationSourceManager:
-    @patch("dispatcher.source.ubuntu_source_manager.move_file")
-    def test_update_app_source_successfully(self, mock_move):
+    def test_add_app_with_gpg_key_successfully(self):
```

Comment:
Add unit tests to verify the behavior of the `UbuntuApplicationSourceManager.add` method when DEB822 format is used. [important]